query-service-impl/src/main/java/org/hypertrace/core/query/service/ExecutionContext.java

```diff
@@ -35,14 +38,66 @@
   // while the selectedColumns
   // is a set of column names.
   private final LinkedHashSet<Expression> allSelections;
+  private final Duration timeSeriesPeriod;
+  private final Duration timeRangeDuration;
+  //  private final String colName;
+  // add request handler in execution context
+  // add another method in interface getStartTimeAttribute
 
   public ExecutionContext(String tenantId, QueryRequest request) {
     this.tenantId = tenantId;
     this.selectedColumns = new LinkedHashSet<>();
     this.allSelections = new LinkedHashSet<>();
+    this.timeSeriesPeriod = setTimeSeriesPeriod(request);
+    this.timeRangeDuration = setTimeRangeDuration(request);
     analyze(request);
   }
 
+  private Duration setTimeSeriesPeriod(QueryRequest request) {
+    Duration period = Duration.ZERO;
+    if (request.getGroupByCount() > 0) {
+      for (Expression expression : request.getGroupByList()) {
+        if (expression.getValueCase() == ValueCase.FUNCTION
+            && expression.getFunction().getFunctionName().equals("dateTimeConvert")) {
+          String periodInSec =
+              expression
+                  .getFunction()
+                  .getArgumentsList()
+                  .get(3)
+                  .getLiteral()
+                  .getValue()
+                  .getString()
+                  .split("[:]")[0];
+          period = Duration.ofSeconds(Long.parseLong(periodInSec));
```

Comment:
we're discarding the unit and assuming it's seconds, but we have it right here - might as well use it and drop the assumption (that is, the split call will return us something like `["30", "SECONDS"]`)